src/services/validation.js

```diff
@@ -1,34 +1,35 @@
 import firebaseAPI from "@/services/API/checkUser.js";
 
-export async function validateSRN(userID, validateCount, isUserValid){
-    // this function is invoked only for the check of SRN's.
-    // validateCount = 0, when the user gets validated on the first try
-    // validateCount = 1, when the user fails the first check
-    // validateCount = 2, when user fails the second check as well -> letting the user go through irrespective.
-    
+/* this function validates the entry against the database (for this usecase, it is firestore). 
+firebaseAPI service is used to validate and it returns a boolean value, indicating whether the user is valid or not.
+If the user is valid and validateCount = 0, then the function returns.
+If the user is not valid and validateCount = 0, then the user is not valid, so validateCount becomes 1 and login message displays an error, allowing the user to correct their entry.
+If the user is not valid and validateCount = 1, then the user has already been authenticated once before and the user has retyped, yet the entry is still invalid, then validateCount becomes 2 to indicate this behaviour.
+@params {String} - userID - current ID being authenticated 
+@params {Number} - validateCount - indicates how many times the user has been validated 
+@params {Boolean} - isCurrentUserValid - holds the value returned by the API */
+export async function validateSRN(userID, validateCount, isCurrentUserValid){
     var invalidLoginMessage = ""
 
-    isUserValid = await firebaseAPI.checkUserExists(userID);
-    if(isUserValid){
+    isCurrentUserValid = await firebaseAPI.checkUserExists(userID);
+    if(isCurrentUserValid){
         return {
-            isUserValid: isUserValid,
+            isCurrentUserValid: isCurrentUserValid,
             validateCount: validateCount,
             invalidLoginMessage: invalidLoginMessage
             }
         }
 
-
-    // this condition checks if the user is getting authenticated the first time. Just shows an error message.
-    if (!isUserValid && validateCount == 0) {
+    if (!isCurrentUserValid && validateCount == 0) {
```

Comment:
you no longer need to check for `!isCurrentUserValid` right? otherwise the code would have stopped in the earlier return statement itself.